src/MemoryBase.cs

```diff
@@ -1,56 +1,67 @@
 using System;
 using System.Text;
 using System.Buffers.Binary;
+using System.Runtime.InteropServices;
 
 namespace Wasmtime
 {
+    /// <summary>
+    /// The base class for memories.
+    /// </summary>
     public abstract class MemoryBase
```

Comment:
Out of curiosity, how come there's a split between this and `Memory`?

Reply:
Before `Memory` was encapsulating an owned `wasm_memory_t` and `CallerMemory` was encapsulating an owned `wasm_extern_t`; `MemoryBase` was just a way to share the implementation between the two.

With the refactoring there's no reason for `CallerMemory` any longer, so I can remove this and consolidate `MemoryBase` with `Memory`.